### PR TITLE
Add job for kind integration tests

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml [hunk 1]
```diff
@@ -181,12 +179,65 @@ tests:
...
+      timeout: 10m0s
+    - as: integration-kind
+      cli: latest
+      commands: |
+        export QUAY_PASS=$(cat /maistra-secrets/password)
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          entrypoint \
+          sh -c \
+          "export TAG=PR-${PULL_NUMBER} && \
+          docker login -u='maistra-dev+prow' -p=${QUAY_PASS} quay.io && \
```
Comment (mattmahoneyrh) [inline ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:216]: Is the docker login needed?

**Review by mattmahoneyrh — COMMENTED** [review]
Is the docker login needed?

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml (hunk 1 above)
Comment (jwendell) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:216]: yes. we build the image and push to quay, similar to the ocp job

**Comment by fjglira** [comment]
@jwendell the kind test it's failing because was not able to join the node to the kind cluster:

```
• Joining worker nodes 🚜  ...
 ✗ Joining worker nodes 🚜
Deleted nodes: ["istio-operator-control-plane" "istio-operator-worker"]
ERROR: failed to create cluster: failed to join node with kubeadm: command "docker exec --privileged istio-operator-worker kubeadm join --config /kind/kubeadm.conf --skip-phases=preflight --v=6" failed with error: exit status 1
Command Output: I1005 20:00:31.712856     200 join.go:412] [preflight] found NodeName empty; using OS hostname as NodeName
I1005 20:00:31.7[129](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44023/rehearse-44023-pull-ci-maistra-istio-operator-maistra-3.0-integration-kind/1710012088534962176#1:build-log.txt%3A129)01     200 joinconfiguration.go:76] loading configuration from "/kind/kubeadm.conf"
I1005 20:00:31.7[135](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44023/rehearse-44023-pull-ci-maistra-istio-operator-maistra-3.0-integration-kind/1710012088534962176#1:build-log.txt%3A135)01     200 controlplaneprepare.go:225] [download-certs] Skipping certs download
I1005 20:00:31.713521     200 join.go:529] [preflight] Discovering cluster-info
```

**Comment by jwendell** [comment]
I know :)

**Pull request merged**